util/time.c

```diff
@@ -0,0 +1,27 @@
+#define _POSIX_C_SOURCE 200809L
+#include <stdint.h>
+#include <time.h>
+
+#include "util/time.h"
+
+const long NSEC_PER_SEC = 1000000000;
+
+uint32_t timespec_to_msec(const struct timespec *a) {
```

Comment:
I'd prefer to have this return an `int64_t`, in case it's called with the result of `timespec_sub`.

Reply:
Ah, that explains why it was signed in the first place -- done.